DROOLS-6943 DMN doc update built-in function DMNv1.4 (#4212)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/dmn/ref-dmn-feel-builtin-functions.adoc

If `keys` is a list of 2 or more elements, this is equivalent of calling `context put(context, key', value')`, with: +
`key'` is the head element in the list keys, +
`value'` is the result of invocation of `context put(context', keys', value)`, where: +
`context'` is the result of `context.key'`, +
`keys'` is the remainder of the list keys without the head element `key'`.

If `keys` is an empty list or null, the result is null.

.Parameters
[cols="30%,70%", options="header"]
|===
|Parameter
|Type

|`context`
|`context`

|`keys`
|`list` of `string`

|`value`
|`Any` type
|===

.Examples
[source,FEEL]
----
context put({x:1}, ["y"], 2) = context put({x:1}, "y", 2)
context put({x:1}, ["y"], 2) = {x:1, y:2}
context put({x:1, y: {a: 0} }, ["y", "a"], 2) = context put({x:1, y: {a: 0} }, "y", context put({a: 0}, ["a"], 2))
context put({x:1, y: {a: 0} }, ["y", "a"], 2) = {x:1, y: {a: 2} }
context put({x:1, y: {a: 0} }, [], 2) = null
----
--
// ----------------------------------------------------------------------------

context merge( contexts )::
+
--

> Returns a new context that includes all entries from the given contexts; if some of the keys are equal, the entries are overriden.

Reviewer: ```suggestion
Returns a new context that includes all entries from the given contexts; if some of the keys are equal, the entries are overridden.
```

The entries are overridden in the same order as specified by the supplied parameter, with new entries added as the last entry in the new context.

.Parameters
[cols="30%,70%", options="header"]
|===
|Parameter
|Type

|`contexts`
|`list` of `context`
|===

.Examples
[source,FEEL]
----
context merge([{x:1}, {y:2}]) = {x:1, y:2}
context merge([{x:1, y:0}, {y:2}]) = {x:1, y:2}
----
--
// ----------------------------------------------------------------------------

== Miscellaneous functions

These functions provide support utilities for several miscellaneous use-cases.
For example, when a decision depends on the current date, like deciding the support SLA over the weekends, additional charges for weekend delivery, etc.

It is important to note that the functions in this section are intended to be side-effect-free, but they are not deterministic and not idempotent from the perspective of an external observer.

As a user, you are encouraged to isolate the decision logic that uses these functions in specific DRG elements, such as Decisions.

> The encapsulation enables them to be overridden with synthetic values during Scenario Testing, that remain constant across executions of the DMN model's test cases.

Reviewer: ```suggestion
The encapsulation enables them to be overridden with synthetic values during scenario testing, that remain constant across executions of the DMN model's test cases.
```

now()::
+
--
Returns the current `date and time`.

.Examples
[source]
----
now()
----
--
// ----------------------------------------------------------------------------

today()::
+
--
Returns the current `date`.

.Examples
[source]
----
today()
----
--
// ----------------------------------------------------------------------------

== KIE Extended functions

> These functions are provided as extension to the DMN Standard to enable support for several miscellaneous use-cases on top of the Drools DMN Engine.

Reviewer: ```suggestion
These functions are provided as an extension to the DMN Standard to enable support for various use-cases on top of the Drools DMN Engine.
```
